pyro/contrib/gp/models/gplvm.py

```diff
@@ -62,80 +56,30 @@ class GPLVM(Parameterized):
         variational parameter ``X_loc``.
     :param str name: Name of this model.
     """
-    def __init__(self, base_model, name="GPLVM"):
-        super(GPLVM, self).__init__(name)
+    def __init__(self, base_model):
+        super(GPLVM, self).__init__()
         if base_model.X.dim() != 2:
             raise ValueError("GPLVM model only works with 2D latent X, but got "
                              "X.dim() = {}.".format(base_model.X.dim()))
         self.base_model = base_model
-        self.y = self.base_model.y
-
-        self.X_loc = Parameter(self.base_model.X)
-
-        C = self.X_loc.shape[1]
-        X_scale_tril_shape = self.X_loc.shape + (C,)
-        Id = eye_like(self.X_loc, C)
-        X_scale_tril = Id.expand(X_scale_tril_shape)
-        self.X_scale_tril = Parameter(X_scale_tril)
-        self.set_constraint("X_scale_tril", constraints.lower_cholesky)
-
-        self._call_base_model_guide = True
+        self.X = Parameter(self.base_model.X)
+        self.set_prior("X", dist.Normal(0, 1).expand(self.X.shape).independent(self.X.dim()))
+        self.autoguide("X", dist.Normal)
```

Comment:
"X" is a parameter with a prior. With `autoguide` method, there is no need to implement `pyro.sample` statements for "X" in `self.model` and `self.guide`.

Reply:
Here the autoguide is `Normal` instead of `MVN`. It is just for simplicity of the model.